src/msg.cpp

```diff
@@ -182,6 +186,33 @@ int zmq::msg_t::init_data (void *data_,
     return 0;
 }
 
+int zmq::msg_t::init_from_allocator (size_t size_, zmq::allocator_t *alloc_)
+{
+    zmq_assert (alloc_ != NULL && size_ != 0);
+
+    _u.lmsg.metadata = NULL;
+    _u.lmsg.type = type_lmsg;
+    _u.lmsg.flags = 0;
+    _u.lmsg.allocator_was_used = 1;
```

Comment:
I don't think we need another field, you can use the ffn and maybe another msg type?

Reply:
yes ok I can do this change by adding a message type "type_lmsg_from_allocator" or something like that.

I was a bit scared by the idea of adding a new msg type since I was not sure if that would trigger some side effect: there are a few methods like zmq::msg_t::is_lmsg()  . I will probably need to find out the places where they get called and understand if I need to add also a call to the new 
   zmq::msg_t::is_lmsg_from_allocator()
but that's doable of course